### ISSUE TITLE
Creating a NaN model prediction happens without warning

### ISSUE THREAD

**Issue by ehultee** [bryanvriel/iceutils#13]
I am now using functionality from `time_series_inversion.ipynb` to separate out signals in Helheim time series.  I was surprised to find that the script ([link](https://github.com/ehultee/Gld-timeseries/blob/master/helheim-tseries_decomp.py)) can run without errors/warnings but generate a totally NaN model prediction.  I looked at the code and there is no built-in `Error` to alert the user what is going wrong...maybe we should add some?

I haven't really poked into this yet, so I'm not sure where we'd warn.  Just flagging it for now.

**Comment by bryanvriel** [comment]
That's most likely due to NaNs being passed to the solver, which doesn't currently do its own screening for valid data. If you look at `iceutils/tseries/solver.py`, starting around [here](https://github.com/bryanvriel/iceutils/blob/d42d51b5c0db400596f784d4163939466b96dd79/iceutils/tseries/solver.py#L152), you can see that some logical indices are created for the valid data to be passed to the solver.

It does make more sense for the solver to handle that internally, but I also don't want to throw an error/warning for every pixel that doesn't have enough valid data. I'll look into ways to do that.

**Comment by ehultee** [comment]
I can't check out the branch and do a full test right now but will try to get to it later today.

Note that passing even one NaN produces a model prediction that is entirely NaN.   For example using `series[3]` in the example script I linked, the length of both data and prediction arrays in the example script is 186, but `sum(np.isnan())` returns 1 for the data and 186 for the prediction.  I don't think this will be fixed by the `n_min` argument for any `n_min != N`

Anyway, thanks for looking into it.  I will get back to you once I've tried "warn"

**Comment by bryanvriel** [comment]
That's right, a single NaN in an array passed to a linear algebra routine will just give you NaNs. The `n_min` argument specifies the minimum number of non-NaN data points needed to proceed with the inversion. In your case, let's say for `series[3]`, as long as `n_min` is less than 185, the inversion will proceed. In that case, the NaN value is removed before the data are sent to the lower-level solvers.

**Comment by ehultee** [comment]
Aha, yeah of course.  I didn't think about that before trying this (lots going on...), but more generally, I think we should assume most real data will have some NaNs.  And out of curiosity, I tried masking the NaN values in the input as a simple fix, but that was unsuccessful.

The PR does seem to solve this exact problem - nice!  Re: your point about not throwing warnings for any single NaN, I agree that would make the code impractical for real data.  I think it can be addressed on the documentation side.  Maybe once EGU is over, I can help write up a documentation notebook that takes real, imperfect data and handles the problems we've uncovered thus far.  We can follow up about that later. :)
